drivers/interrupt_controller/intc_ioapic.c

```diff
@@ -128,7 +128,7 @@ static uint16_t ioapic_id;
 
 static bool get_vtd(void)
 {
-	union acpi_dmar_id *dmar_id;
+	union acpi_dmar_id dmar_id;
 	int inst_cnt;
 
 	if (vtd != NULL) {
```

Comment:
in addition to your changes, this should be:
```
if (!device_is_ready(vtd)) {
        return false;
}

if (ioapic_id != 0) {
        return true;
}
```

make sure ioapic_id is 0 at boot time (it should be since it's in the .bss section afaik)